instant-segment/src/lib.rs

```diff
@@ -40,12 +39,19 @@ impl Segmenter {
     /// The `HashMap` type here refers to `std::collections::HashMap` parametrized with the
     /// `ahash::RandomState`.
     pub fn from_maps(
-        unigrams: HashMap<String, f64>,
-        bigrams: HashMap<(String, String), f64>,
+        mut unigrams: HashMap<String, f64>,
+        mut bigrams: HashMap<(String, String), f64>,
     ) -> Self {
+        let uni_total: f64 = unigrams.values().sum();
```

Comment:
Style nit: I don't think the type annotations are actually necessary here? I usually prefer to avoid them in favor of turbofishing because for the latter it's usually more obvious how/where it is needed.